lib/BloqadeExpr/src/linalg.jl

```diff
@@ -5,3 +5,66 @@ function LinearAlgebra.mul!(C::AbstractVecOrMat, A::StepHamiltonian, B::Abstract
     end
     return C
 end
+
+# if BloqadeExpr was the backend of choice, then ThreadedMatrix will have the SparseMatrixCSC type
+# in which case we just dispatch to standard mul!
+LinearAlgebra.mul!(C, A::ThreadedMatrix{<:SparseMatrixCSC}, B, α, β) = SparseArrays.mul!(C, A.matrix, B, α, β) 
```

Comment:
I think we need some basica tests for these

Reply:
Can do!